src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -482,16 +485,40 @@ def _get_data(
                 # This should only happen for payment status update,
                 # where only a subset of the variables are updated.
                 continue
-            target_path = mapping["target_path"]
 
-            # Type hint is wrong: currently some static variables are of type date/datetime
             value = cast(Any, variables_values[variable_key])
 
             # Comply with JSON Schema "format" specs:
             if isinstance(value, (datetime, date)):
                 value = value.isoformat()
 
-            glom.assign(record_data, glom.Path(*target_path), value, missing=dict)
+            target_path = mapping.get("target_path")
+            if (
+                component_key_type_mappings
+                and component_key_type_mappings.get(variable_key) == "addressNL"
+            ):
+                assert isinstance(value, dict)
+                filtered_value = {k: v for k, v in value.items() if v}
+                if target_path:
+                    value.pop("secretStreetCity")
+                    glom.assign(
+                        record_data,
+                        glom.Path(*target_path),
+                        filtered_value,
+                        missing=dict,
+                    )
+
+                elif subfields_mapping := mapping.get("options"):
+                    for subfield_key, subfield_value in subfields_mapping.items():
+                        glom.assign(
+                            record_data,
+                            glom.Path(*subfield_value),
+                            filtered_value[underscore_to_camel(subfield_key)],
```

Comment:
I'd rather we don't have to deal with snake/camel case conversion, but that would require re-structuring the data model from a simple key-value object to an array of mapping objects, which will impact the frontend code again. So, maybe just put a `# TODO` here to remark that and let's leave it as it is then

Reply:
what's more concerning, is that `filtered_value` excludes empty-ish values, but the `subfield_key` access doesn't consider that. E.g. if you only fill out postal code + house number and leave house letter empty, while house letter is mapped, this will crash. So I'd definitely write a test for this case too :)